src/components/BaseAvatar.vue

```diff
@@ -1,26 +1,26 @@
-<script setup>
+<script setup lang="ts">
 import { watch, ref } from 'vue';
 
-const props = defineProps({
-  address: String,
-  size: String,
-  imgsrc: String,
-  space: Object
-});
+const props = defineProps<{
+  address?: string;
```

Comment:
Instead of keeping all props optional its better to define different combinations that is possible with this. 

`{address: string; space: Record<String, any>} | {imgsrc: string; space: Record<String, any>}`

Reply:
Doesn't work for me 
<img width="909" alt="image" src="https://user-images.githubusercontent.com/51686767/156924578-e7998d3b-a2a2-4d38-a26c-193e0c1687ba.png">

Reply:
Anyway they are all optional, we can leave it like this